From 70939b6cf8820330d7969e1621d36e0527fea545 Mon Sep 17 00:00:00 2001
From: Kiyoshi Shikuma <kiyoshi@brilliant.tech>
Date: Wed, 27 Apr 2022 12:26:55 -0700
Subject: [PATCH 2/2] Revert "rtspsrc: append aggregate control string to base
 URL before query string"

This reverts commit ed6b5a3a632807ea00ae7c609f0c0bed69a9078e.
---
 ../gst/rtsp/gstrtspsrc.c    | 27 +++++++------------
 1 file changed, 10 insertions(+), 17 deletions(-)

diff --git a/gst/rtsp/gstrtspsrc.c b/gst/rtsp/gstrtspsrc.c
index 396d2b5174..48363f1e04 100644
--- a/gst/rtsp/gstrtspsrc.c
+++ b/gst/rtsp/gstrtspsrc.c
@@ -2375,28 +2375,21 @@ gst_rtspsrc_create_stream (GstRTSPSrc * src, GstSDPMessage * sdp, gint idx,
     if (g_str_has_prefix (control_url, "rtsp://"))
       stream->conninfo.location = g_strdup (control_url);
     else {
+      const gchar *base;
+      gboolean has_slash;
+
       if (g_strcmp0 (control_url, "*") == 0)
         control_url = "";
-      /* handle url with query */
-      if (src->conninfo.url && src->conninfo.url->query) {
-        stream->conninfo.location =
-            gst_rtsp_url_get_request_uri_with_control (src->conninfo.url,
-            control_url);
-      } else {
-        const gchar *base;
-        gboolean has_slash;
 
-        base = get_aggregate_control (src);
+      base = get_aggregate_control (src);
 
-        /* check if the base ends or control starts with / */
-        has_slash = g_str_has_prefix (control_url, "/");
-        has_slash = has_slash || g_str_has_suffix (base, "/");
+      /* check if the base ends or control starts with / */
+      has_slash = g_str_has_prefix (control_url, "/");
+      has_slash = has_slash || g_str_has_suffix (base, "/");
 
-        /* concatenate the two strings, insert / when not present */
-        stream->conninfo.location =
-            g_strdup_printf ("%s%s%s", base, has_slash ? "" : "/", control_url);
-
-      }
+      /* concatenate the two strings, insert / when not present */
+      stream->conninfo.location =
+          g_strdup_printf ("%s%s%s", base, has_slash ? "" : "/", control_url);
     }
   }
   GST_DEBUG_OBJECT (src, " setup: %s",
-- 
2.32.0 (Apple Git-132)

